### PR TITLE
feat(style-linter): adds addition var rules, updates order rules and audit codebase

### PR DESCRIPTION
**Pull request by mcheng09**
# Description

Added the following rules below: 

- Color and background only accepts $variables. i.e. `color: $purple`

- Media queries only accept $variables. i.e. `@media(max-width: $screen-sm)`
 
- No !important tags allowed!
 
- @extends must be placed before all declarations

- @media queries must be placed after declaration before anything else

Updated the following rules below: 

- Changed properties orders to follow the convention below
https://github.com/hudochenkov/stylelint-config-hudochenkov/blob/master/order.js

- BEM linter now accepts ${i} format for SASS loops

Removed the following rules below: 

- Idiomatic SCSS extension

### CODE REVIEW

File: src/components/Footer/Footer.scss [hunk 1]
```diff
@@ -1,27 +1,32 @@
 @import '../../styles/utilities/colors';
+@import '../../styles/utilities/breakpoints';
```
Comment (mattlaguardia) [inline src/components/Footer/Footer.scss:2]: nice catch thank you for updating these

File: src/styles/utilities/typography.scss [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 
 h4 {
-  font-family: $poppins;
-  font-size: 21px;
   font-weight: 500;
-  letter-spacing: 0.02em;
+  font-size: 21px;
+  font-family: $poppins;
   line-height: 31px;
+  letter-spacing: 0.02em;
```
Comment (mattlaguardia) [inline src/styles/utilities/typography.scss:14]: this file i think should be double checked with the designs

Comment (mattlaguardia) [reply src/styles/utilities/typography.scss:14]: ie idk if design has letters spaced anywhere

File: stylelint.config.js [hunk 3]
```diff
@@ -27,9 +30,260 @@ module.exports = {
...
+				"margin-top",
+				"margin-right",
+				"margin-bottom",
+				"margin-left",
+				"padding",
+				"padding-top",
+				"padding-right",
+				"padding-bottom",
+				"padding-left",
+				"border",
```
Comment (mattlaguardia) [inline stylelint.config.js:118]: Border is before padding

File: stylelint.config.js [hunk 4]
```diff
@@ -27,9 +30,260 @@ module.exports = {
...
+				"background-image",
+				"background-attachment",
+				"background-position",
+				"background-position-x",
+				"background-position-y",
+				"background-clip",
+				"background-origin",
+				"background-size",
+				"background-repeat",
+				"color",
```
Comment (mattlaguardia) [inline stylelint.config.js:173]: this should be with fonts IMO

File: stylelint.config.js [hunk 5]
```diff
@@ -27,9 +30,260 @@ module.exports = {
...
+				"outline-offset",
+				"table-layout",
+				"caption-side",
+				"empty-cells",
+				"list-style",
+				"list-style-position",
+				"list-style-type",
+				"list-style-image",
+
+				//
```
Comment (mattlaguardia) [inline stylelint.config.js:189]: i think youre missing comment

File: src/components/Container/Container.scss [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
+@import './../../styles/utilities/alignment.scss';
 
 $total-columns: 12;
 
 .container {
   display: grid;
   max-width: $breakpoint-laptop-large;
-  padding: 5px;
   margin: 0 16px;
+  padding: 5px;
```
[the remark this answers is not in the document]
Comment (mcheng09) [reply src/components/Container/Container.scss:11]: I'll delete it

File: src/components/Container/Container.scss [hunk 7]
```diff
@@ -1,14 +1,14 @@
-@import './../../styles/fonts.scss';
-@import './../../styles/breakpoints.scss';
-@import './../../styles/alignment.scss';
+@import './../../styles/utilities/fonts.scss';
```
[the remark this answers is not in the document]
Comment (mcheng09) [reply src/components/Container/Container.scss:1]: Fonts moved to directory below styles.

File: src/styles/utilities/typography.scss (hunk 2 above)
Comment (mcheng09) [reply src/styles/utilities/typography.scss:14]: As of now the typography is a bit inconsistent and most of it is being overwritten. Might need to bring this up to designer. I can delete the letter-spacing for now.

File: stylelint.config.js (hunk 3 above)
Comment (mcheng09) [reply stylelint.config.js:118]: Border and all it's properties were moved up.

File: stylelint.config.js (hunk 4 above)
Comment (mcheng09) [reply stylelint.config.js:173]: i'm okay with how it is now, but let me know if you do want a change.

File: stylelint.config.js (hunk 5 above)
Comment (mcheng09) [reply stylelint.config.js]: Removing these.

File: stylelint.config.js (hunk 4 above)
Comment (mattlaguardia) [reply stylelint.config.js:169]: lets move it down

File: src/styles/utilities/typography.scss (hunk 2 above)
Comment (mattlaguardia) [reply src/styles/utilities/typography.scss]: works for me

File: stylelint.config.js [hunk 8]
```diff
@@ -27,9 +30,248 @@ module.exports = {
...
+				"outline-style",
+				"outline-color",
+				"outline-offset",
+				"table-layout",
+				"caption-side",
+				"empty-cells",
+				"list-style",
+				"list-style-position",
+				"list-style-type",
+				"list-style-image",
```
[the remark this answers is not in the document]
Comment (mcheng09) [reply stylelint.config.js:183]: Thanks for the reference 👍

**Pull request merged**